recipes/dcase2021_task4_baseline/finetune_on_separated.py

```diff
@@ -0,0 +1,364 @@
+        audio_len=config["data"]["audio_max_len"],
+        frame_len=config["feats"]["n_filters"],
+        frame_hop=config["feats"]["hop_length"],
+        net_pooling=config["data"]["net_subsample"],
+        fs=config["data"]["fs"],
+    )
+
+    devtest_df = pd.read_csv(config["data"]["test_tsv"], sep="\t")
+    devtest_dataset = StronglyAnnotatedSet(
+        config["data"]["test_folder_sep"],
+        devtest_df,
+        encoder,
+        return_filename=True,
+        pad_to=config["data"]["audio_max_len"],
+        multisrc=True,
+    )
+
+    test_dataset = devtest_dataset
+
+    ##### model definition  ############
+
+    ### loading pre-trained SED model ###########
+    with open(config["training"]["sed_yaml"], "r") as f:
+        sed_yaml = yaml.load(f)
+
+    pretrained = CRNN(**sed_yaml["net"])
+    sed_trainer = SEDTask4_2021(
+        sed_yaml,
+        encoder=encoder,
+        sed_student=pretrained,
+        opt=None,
+        train_data=None,
+        valid_data=None,
+        test_data=None,
+        train_sampler=None,
+        scheduler=None,
+    )
+
+    ckpt = torch.load(config["training"]["sed_checkpoint"], map_location="cpu")
+    sed_trainer.load_state_dict(ckpt["state_dict"])
+
+    sed_model = CRNN(**config["net"], freeze_bn=True)  # freezing batch norm
+    if config["training"]["sed_model"] == "student":
+        sed_model.load_state_dict(sed_trainer.sed_student.state_dict(), strict=False)
+    elif config["training"]["sed_model"] == "teacher":
+        sed_model.load_state_dict(sed_trainer.sed_teacher.state_dict(), strict=False)
+    else:
+        raise EnvironmentError
+
+    sed_model = EnsembleModel(sed_model)
+
+    if test_state_dict is None:
+        ##### data prep train valid ##########
+        synth_df = pd.read_csv(config["data"]["synth_tsv"], sep="\t")
+        synth_set = StronglyAnnotatedSet(
+            config["data"]["synth_folder_sep"],
+            synth_df,
+            encoder,
+            pad_to=config["data"]["audio_max_len"],
+            multisrc=True,
+        )
+
+        weak_df = pd.read_csv(config["data"]["weak_tsv"], sep="\t")
+        train_weak_df = weak_df.sample(
+            frac=config["training"]["weak_split"],
+            random_state=config["training"]["seed"],
+        )
+        valid_weak_df = weak_df.drop(train_weak_df.index).reset_index(drop=True)
+        train_weak_df = train_weak_df.reset_index(drop=True)
+        weak_set = WeakSet(
+            config["data"]["weak_folder_sep"],
+            train_weak_df,
+            encoder,
+            pad_to=config["data"]["audio_max_len"],
+            multisrc=True,
+        )
+
+        unlabeled_set = UnlabelledSet(
+            config["data"]["unlabeled_folder_sep"],
+            encoder,
+            pad_to=config["data"]["audio_max_len"],
+            multisrc=True,
+        )
+
+        synth_df_val = pd.read_csv(config["data"]["synth_val_tsv"], sep="\t")
+        synth_val = StronglyAnnotatedSet(
```

Comment:
Just realising we have inconsistency between names:
- WeakSet
- Unlabelled set --> Unlabeled set ? (typo?)
- StronglyAnnotatedSet

Maybe we could have:
- WeaklyLabeledSet
- UnlabeledSet
- StronglyLabeledSet
Could be done in a future pull request I guess.